src/PostgREST/Preferences.hs

```diff
@@ -0,0 +1,61 @@
+{-|
+Module      : PostgREST.Types
+Description : PostgREST common types and functions used by the rest of the modules
+-}
+{-# LANGUAGE DuplicateRecordFields #-}
+
+module PostgREST.Preferences where
```

Comment:
So this seems to actually "belong" into `ApiRequest`, I think. But I see, that this would just make the file quite long.

What do you think about:
- `PostgREST.Request.Preferences` - basically this file (+ maybe some of the "Parsers" / read equivalents we have in `ApiRequest`)
- `PostgREST.Request` (renamed from `ApiRequest`): this could re-export the Preferences or we could just import from `PostgREST.Request.Preferences`.

We can probably/hopefully split up some more stuff in the `Request` namespace later.